internal/rootcoord/meta_table.go

```diff
@@ -1291,6 +1293,32 @@ func (mt *MetaTable) AddCredential(credInfo *internalpb.CredentialInfo) error {
 	if credInfo.Username == "" {
 		return fmt.Errorf("username is empty")
 	}
+	mt.permissionLock.Lock()
+	defer mt.permissionLock.Unlock()
+
+	usernames, err := mt.catalog.ListCredentials(mt.ctx)
+	if err != nil {
+		return err
+	}
+	if len(usernames) >= Params.ProxyCfg.MaxUserNum {
+		return errors.New("unable to add user because the number of users has reached the limit")
```

Comment:
logging max user number

Reply:
It will be added in my next pull request.